### ISSUE TITLE
Request: A way to ask the kernel for the Type that would be used to inject a certain service

### ISSUE THREAD

**Issue by havremunken** [ninject/Ninject#115]
WCF services need to have a parameterless constructor for the system to instantiate them. Of course, I have services that I want to inject services into via constructor injection. I use Ninject 3 and have created custom NinjectServiceHost, NinjectInstanceProvider and NinjectServiceBehavior classes. This means that when I instantiate a ServiceHost, it gets injected with the necessary stuff from my configured IKernel.

However, right now I have to do this in a super-ugly way, because the constructor for a ServiceHost takes a type as an argument - the type of the service to instantiate. Not the service contract interface, but the concrete type. 

So if I have an ISomeService, I would love to ask the kernel and get a response that would be the Type object for SomeServiceImpl for instance.

Basically, do everything but actually instantiate the concrete type, and just tell me the type instead.

Right now, to "cheat", I have to ask for a concrete implementation and run GetType() on it to find my answer, and I need to take a shower every time I look at my source code.

Would this be possible?
